stan/math/prim/scal/fun/constants.hpp

```diff
@@ -11,92 +11,169 @@ namespace stan {
 namespace math {
 
 /**
- * The base of the natural logarithm,
- * \f$ e \f$.
+ * Return the base of the natural logarithm.
+ *
+ * @return Base of natural logarithm.
  */
-const double E = boost::math::constants::e<double>();
+inline double e() { return boost::math::constants::e<double>(); }
 
 /**
- * The value of the square root of 2,
- * \f$ \sqrt{2} \f$.
+ * Return the value of pi.
+ *
+ * @return Pi.
  */
-const double SQRT_TWO = std::sqrt(2.0);
+inline double pi() { return boost::math::constants::pi<double>(); }
```

Comment:
Can you add a note to these that when we move up to C++17 these can be constexpr

Reply:
Done.